src/Raven.Server/Documents/Handlers/Processors/Replication/ReplicationHandlerProcessorForGetOutgoingFailureStats.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Raven.Server.Web.Http;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers.Processors.Replication
+{
+    internal class ReplicationHandlerProcessorForGetOutgoingFailureStats : AbstractReplicationHandlerProcessorForGetOutgoingFailureStats<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public ReplicationHandlerProcessorForGetOutgoingFailureStats([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override bool SupportsCurrentNode => true;
+
+        protected override async ValueTask HandleCurrentNodeAsync()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            await using (var writer = new AsyncBlittableJsonTextWriterForDebug(context, ServerStore, RequestHandler.ResponseBodyStream()))
+            {
+                var data = new DynamicJsonArray();
+                foreach (var item in RequestHandler.Database.ReplicationLoader.OutgoingFailureInfo)
+                {
```

Comment:
I mean to have something like
```
["Key"] = item.Key.ToJson(),
["Value"] = item.Value.ToJson()
```
In general those dictionaries are problematic, because the Key is a nested object.
Wonder if we can/should do something about that.. @ppekrol ?

Reply:
done